libqtile/widget/systray.py

```diff
@@ -150,6 +150,25 @@ def _configure(self, qtile, bar):
         )
         qtile.core._root.send_event(event, mask=EventMask.StructureNotify)
 
+        # Bit of an ugly hack here
+        # Don't immediately see why this shouldn't work for 24 bit depth visuals...
+        if self.drawer.depth == 32:
+
+            # If we're using transparency then We need tray to tell icons which
+            # visual to use
+            tray_attrs = self.conn.conn.core.GetWindowAttributes(win.wid).reply()
+            tray_visual = tray_attrs.visual
```

Comment:
What value does `tray_visual` have? Will it always be whatever we specified when the window was created?

Reply:
Are you trying to point out that I could have just pulled that value from the `drawer`...?

Reply:
Only if the answer is yes :laughing: